sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -61,7 +61,7 @@ object IntervalUtils {
   }
 
   def getDays(interval: CalendarInterval): Long = {
```

Comment:
We need to redefine the semantic here. To exact days from `INTERVAL 1 day 48 hour`, previously we return 3 days, now we return 1 day. Which one is corrected? cc @MaxGekk

Reply:
Same question to `getHours`. For now it's not consistent with `getDays`

Reply:
The method was added for feature parity with PostgreSQL which tracks days in a separate variable too, and doesn't perform normalization:
```sql
maxim=# select INTERVAL '1 day 48 hour';
    interval
----------------
 1 day 48:00:00
(1 row)
```
> Which one is corrected?

I think we could change current behavior.

Reply:
I played with pgsql a little bit
```
cloud0fan=# SELECT EXTRACT(DAY FROM INTERVAL '1 days 50 hours');
 date_part 
-----------
         1
(1 row)

cloud0fan=# SELECT EXTRACT(HOUR FROM INTERVAL '1 days 50 hours');
 date_part 
-----------
        50
(1 row)

cloud0fan=# SELECT EXTRACT(HOUR FROM INTERVAL '1 days 50 hours 100 minutes');
 date_part 
-----------
        51
(1 row)
```

So `getDays` should return the `days` field directly. `getHours` should simply return `microseconds / MICROS_PER_HOUR`. @LinhongLiu can you fix it?